### PR TITLE
A few optimizations

### PR DESCRIPTION
**Pull request by jaredlewis**
Use iterator and add update entity_kind to be entity_kind_id @wesleykendall

### CODE REVIEW

File: entity/sync.py [hunk 1]
```diff
@@ -95,14 +95,22 @@ def _sync_all_entities(self):
...
+                entity_ids.append(model_obj.id)
+
+                # Sync the entity
                 self._sync_entity(model_obj)
 
             # Delete any existing entities that are not in the model obj table
             Entity.all_objects.filter(entity_type=ContentType.objects.get_for_model(
                 entity_model, for_concrete_model=False)).exclude(
-                entity_id__in=(model_obj.id for model_obj in model_objs)).delete(force=True)
+                entity_id__in=entity_ids).delete(force=True)
```
Comment (wesleykendall) [inline entity/sync.py]: I'm failing to see how this optimizes things. An iterator is invalid when using .prefetch_related, which is what we use in virtually all of our entity querysets. Along with that, all of the entities that are passed to _sync_entity are being stored in an internal cache, so this doesn't save any memory. Building the ID list in the for loop is also less efficient than the generator you removed.

Can you elaborate more on this optimization? I'm not sure what it's doing

**Comment by wesleykendall** [comment]
@jaredlewis had some comments

**Comment by jaredlewis** [comment]
In solarcitys case using the iterator was close to 10 seconds faster, even with the extra prefetch queries. I build the id list in the loop because im using the iterator and dont want to loop over it twice

**Comment by jaredlewis** [comment]
Also the memory footprint using iterator is 250MB less using the iterator in this use case

**Comment by jaredlewis** [comment]
Im ok with not using the iterator, but we do need a way to not load all the entities into memory at once

**Pull request merged**